### PR TITLE
[Feature:Calendar] Calendar Color Select + Legend

### PR DESCRIPTION
**Pull request by mahi-pas**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?
Students couldn't select colors or see a legend where all the colors are displayed

### What is the new behavior?
Students can select what color the course is displayed as well as see a legend
image
image

Additionally, there are 8 color options for light and dark modes. These colors all pass WAVE's guidelines for contrast with text.
These are calendars with all colors displayed:
Screenshot from 2023-08-14 15-54-17
Screenshot from 2023-08-14 15-33-11
Screenshot from 2023-08-14 15-33-43
*Note: Due to changing color options, this has also changed the colors used in other places such as docker container colors. Please check to make sure nothing has broken.*

### Other information?
Yet to add:
Using course names rather than codes

### CODE REVIEW

**Comment by mahi-pas** [comment]
For colors, I'm going to test each category color using this jquery function
```js
$('.cal-gradeable-item').each(function(i) {
    $(this).css('background-color', `var(--category-color-${i})`);
})
```

**Review by 04vmatsibekker — CHANGES_REQUESTED** [review]
Looks pretty good. One thing that I will say though, is that we are unable to distinguish course notes by color. As you can see in the image below, the announcement in the class blank, has a colored border, but the blank note has no color indication. Is there any that you can maybe add? Or do you think this would be a separate PR. 
image

**Review by NiaHeermance — APPROVED** [review]
Looks great!!! I wish yellow was a color though, ha ha. Approved!

**Review by ziesski — APPROVED** [review]

File: site/app/templates/admin/Docker.twig [hunk 1]
```diff
@@ -139,7 +139,7 @@
                 </td>
                 <td>
                     {% for capability in value %}
-                        <span class="badge badge-{{ capability_to_color[capability] }}">{{capability}}</span>
+                        <span class="badge badge-{{ capability_to_color[capability] % 12 }}">{{capability}}</span>
```
[the remark this answers is not in the document]
Comment (mahi-pas) [reply site/app/templates/admin/Docker.twig:142]: This modulo is making sure that it doesn't go past 12 colors. Although I should change it to 8. I did this so we can reduce the number of category colors down the line

**Review by 04vmatsibekker — APPROVED** [review]

**Pull request merged**